torchtune/utils/metric_logging.py

```diff
@@ -160,27 +174,63 @@ def __init__(
                 "Alternatively, use the ``StdoutLogger``, which can be specified by setting metric_logger_type='stdout'."
             ) from e
         self._wandb = wandb
-        self._wandb.init(
-            project=project,
-            entity=entity,
-            group=group,
-            reinit=True,
-            resume="allow",
-            config=kwargs,
-        )
+
+        # logging strategy options are "main", "node", "all"
+        self.log_strategy = log_strategy
+        self.world_size, self.rank = get_world_size_and_rank()
+        self.local_rank = _get_local_rank()
```

Comment:
Probably worth integrating these changes into e.g. `full_finetune_distributed.py` as a test of this logic. Cause on single-device recipe these will all be equivalent, right?